test/suite/test_verbose03.py

```diff
@@ -0,0 +1,175 @@
+    # The JSON schema we expect all messages to follow. Captures all possible fields, detailing
+    # each field's name, associated type and whether we always expect for that field to be
+    # present.
+    expected_json_schema = {
+        'ts_sec': {'type': int, 'always_expected': True },
+        'ts_usec': {'type': int, 'always_expected': True },
+        'thread': {'type': str, 'always_expected': True },
+        'session_err_prefix': {'type': str, 'always_expected': False },
+        'session_dhandle_name': {'type': str, 'always_expected': False },
+        'session_name': {'type': str, 'always_expected': False },
+        'error_str': {'type': str, 'always_expected': False },
+        'error_code': {'type': int, 'always_expected': False },
+        'msg': {'type': str, 'always_expected': True },
+    }
+
+    # Validates the JSON schema of a given event handler message, ensuring the schema is consistent and expected.
+    def validate_json_schema(self, json_msg):
+        expected_schema = dict(self.expected_json_schema)
+        for field in json_msg:
+            # Assert the JSON field is valid and expected.
+            self.assertTrue(field in expected_schema, 'Unexpected field "%s" in JSON message: %s' % (field, str(json_msg)))
+            # Assert the type of the JSON field is expected.
+            self.assertEqual(type(json_msg[field]), expected_schema[field]['type'],
+                    'Unexpected type of field "%s" in JSON message, expected "%s" but got "%s": %s' % (field,
+                        str(expected_schema[field]['type']), str(type(json_msg[field])), str(json_msg)))
+            expected_schema.pop(field, None)
+        # Go through the remaining fields in the schema and ensure we've seen all the fields that are always expected be present
+        # in the JSON message
+        for field in expected_schema:
+            self.assertFalse(expected_schema[field]['always_expected'], 'Expected field "%s" in JSON message, but not found: %s' %
+                (field, str(json_msg)))
+
+    @contextmanager
+    def expect_event_handler_json(self, config, stdErr=False):
+        # Clean the stdout/stderr resource before yielding the context to the execution block. We only want to
+        # capture the verbose output of the using context (ignoring any previous output up to this point).
+        if stdErr:
+            self.cleanStderr()
+        else:
+            self.cleanStdout()
+        # Create a new connection with JSON format enabled.
+        if stdErr:
+            conn_config = 'json_output=[error]'
+        else:
+            conn_config = 'json_output=[message]'
+        if config != "":
+            conn_config += "," + config
+        conn = self.wiredtiger_open(self.home, conn_config)
+        # Yield the connection resource to the execution context, allowing it to perform any necessary
+        # operations on the connection (for generating the expected message output).
+        yield conn
```

Comment:
Just a python question: What is contextmanager? and how is that utilised with yield? 

From what I see is that we can use the connection object once we yield, and perform any operations. Once those operations are done, then we continue the from here. Is that correct?

Reply:
Yep, that's pretty much correct. The `contextmanager` pattern is essentially used to set up a  resource, yield it to a callers execution context and then once the callers context has ended, the manager is able handle the clean up of that resource. The context manager pattern is commonly seen with examples around opening files e.g. `with open(file, 'r') as myfile:`, where the `contextmanager` manages both the set up of the resource (i.e. opening the file) and cleaning up of the resource (i.e. closing the file).

Specifically we use it here to create an appropriately configured connection object and yield it to the caller's execution _context_ to do whatever activity it wants with that connection. After the callers context has finished, it returns to the `contextmanager`, which lets the manager do the checks it needs and close up the connection. 
I mainly employed this pattern in the verbose python tests to isolate the execution contexts of the various operations, allowing  us to test against the verbose output of only the operations performed in the callers execution context.